Upgrade PlantUML version to 1.2023.10

lesterong opened the pull request:
**What is the purpose of this pull request?**

- [x] Others, please explain: Dependency Update

<!--
  If this pull request is addressing an issue, link to the issue: #xxx

  If this pull request completely addresses an issue, use one of the closing keywords: "Fixes #xxx" or "Resolves #xxx"
  https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

  Otherwise, elaborate further on the rationale of this pull request as needed
-->

**Overview of changes:**
- Upgrade PlantUML to version 1.2023.10 (Fixes #1792)
- Update corresponding images generated by PlantUML
- Adjusted the annotations example in user guide due to the change in theme to the latest PlantUML version

Left: Before
Middle: After 
Right: After, annotation positions adjusted
<img width="927" alt="image" src="https://github.com/MarkBind/markbind/assets/84223259/95d1e8b7-bbb6-441b-9bd9-f2143d6a1967">


Top: Before 
Middle: After 
Bottom: After, annotation positions adjusted
<img width="511" alt="image" src="https://github.com/MarkBind/markbind/assets/84223259/cc96e1db-a7ce-4445-b13b-2885db3e1abd">



**Anything you'd like to highlight/discuss:**
- There is a slight layout shift as the default theme has now changed in this version of PlantUML.
- There is also a change in the diagram generated by `class.puml` as PlantUML has changed how they handle namespaces. This behaviour can be disabled using `set separator none`.


Left: Before 
Right: After
<img width="850" alt="image" src="https://github.com/MarkBind/markbind/assets/84223259/de371eb2-dac9-438b-8e1b-0a6735c1988b">

- This update seems to have broken the `usecase.puml` diagram as it no longer allows for two elements with the same name, though this can be fixed by providing an alias.

<img width="418" alt="image" src="https://github.com/MarkBind/markbind/assets/84223259/1e5c7e04-24bb-402a-b622-e3b79fd29e49">

- There seems to be a change in font and font size as well, so the Gantt Chart text no longer fits in the box.

<img width="414" alt="image" src="https://github.com/MarkBind/markbind/assets/84223259/5885d087-7255-4602-a548-67b0e56311d2">


**Proposed commit message: (wrap lines at 72 characters)**
Bump PlantUML to 1.2023.10

<!--
  See this link for more info on how to write a good commit message:
  https://se-education.org/guides/conventions/git.html

  As best as possible, write a succinct commit title in 50 characters

|---------This is the width of 50 chars----------|
|-----------This is the width of 72 chars for your reference-----------|
-->

---

**Checklist:** :ballot_box_with_check:

<!-- Leave non-applicable items unchecked -->

- [ ] Updated the documentation for feature additions and enhancements
- [ ] Added tests for bug fixes or features
- [x] Linked all related issues
- [x] No unrelated changes <!-- It's tempting, but increases the reviewer's work, and really pollutes the commit history =( -->

<!--
  We'll try our best to get to your PR within a week.
  If we haven't gotten to it then, or if your pull request resolves an urgent item, feel free to give us a ping!
-->

---

### Migration Guide

The PlantUML plugin has been updated from 1.2020.7 to 1.2023.10.

**Changes**
(For detailed changes, please view them on [plantuml website](https://plantuml.com/news))
1. The default theme has been updated in this version of PlantUML. To use the old theme, use the [`skin rose` directive](https://www.plantuml.com/plantuml/uml/SoWkIImgAStDuIhEpinJACelJkLooazIqBLJSCp9J4vLi5B8ICt9oLV8pqbHA4ijAixCp-DoICrB0Me10000).

<img width="400" src="https://github.com/MarkBind/markbind/assets/84223259/8a39b5f0-4032-4867-bd83-b30d968e040a" alt="theme change">

2. There is a slight shift in the layout with the new default theme, and the position of annotations on PlantUML diagrams may have to be updated.
3. PlantUML now separates namespaces by default. This behavior can be disabled using the [`set separator none` command](https://www.plantuml.com/plantuml/uml/5Sl12O0m30NG-v_YJr3G2Jf6hq4Z26qYJJo9kull_cgapBn71qAJePTCIPyqDyNM9O9jAMqjXpiVaE53UThXU86gjl_v0m00).

<img width="400" src="https://github.com/MarkBind/markbind/assets/84223259/5d8d5bd7-0541-4b4e-823e-84a9973f15b6" alt="separators change">

4. PlantUML no longer allows for two elements with the same name. A possible workaround is to provide an alias to one of the elements, so their names do not conflict.

```puml
'alias as c provided
rectangle checkout as c {
  (checkout) .> (payment) : include
}
```

<img width="600" src="https://github.com/MarkBind/markbind/assets/84223259/a4349323-c8f8-421c-8430-06af594d7a25" alt="element with same name alias">


5. PlantUML does not render underlines in class diagrams using the Creole syntax. A possible workaround is to use the HTML syntax to define underlines. For example, `<u>Class Name</u>` will render an underline, but not `__Class Name__`.

<img width="600" src="https://github.com/MarkBind/markbind/assets/84223259/bb8c48a5-e78c-4cd7-ae0a-9edb98b0f2e6" alt="Creole Syntax vs HTML Syntax underline">


6. The font used in Gantt Charts has been updated. As a result, some of the text may appear outside of the bars.

<img width="700" src="https://github.com/MarkBind/markbind/assets/84223259/6bcd2790-baef-4cfd-b5d2-54d3561016e6" alt="Gantt charts change">

lesterong commented:
> _Before_ should_ be like this (as per https://markbind.org/userGuide/components/imagesAndDiagrams.html#diagrams)? ![image](https://user-images.githubusercontent.com/1673303/247175833-10c81294-a3bd-414b-bb84-d9f6c9e277ad.png)

Thanks @damithc, you are right 🙂 . I have since updated my comment to be clearer and better reflect the changes.

itsyme commented:
hi @lesterong! thanks for the work! perhaps it would be good to check out how the [cs2103 website](https://github.com/nus-cs2103-AY2223S2/website) would look after the bump as it is one of our most content heavy websites which is heavily used

yucheng11122017 commented:
Hi @lesterong thanks so much for the work! It looks p good. 
Minor nit: Do you know why the text in docs/userGuide/diagrams/gantt.png "Hire test writers" is outside the box? Is it because of an sizing issue?

lesterong commented:
> Hi @lesterong thanks so much for the work! It looks p good. Minor nit: Do you know why the text in docs/userGuide/diagrams/gantt.png "Hire test writers" is outside the box? Is it because of an sizing issue?

Hi @yucheng11122017, yes that is correct. It appears that PlantUML is now using a different font, and it no longer fits in the bar. I tried increasing the length of the 'Hire test writers' bar, and the text fits in the bar as expected.

yucheng11122017 reviewed (approved):
LGTM! 
Minor point for anyone reviewing: the images in the deploy preview are still the old ones since absolute links are used instead of relative ones. 
May need to update this as well

lesterong commented:
> LGTM! Minor point for anyone reviewing: the images in the deploy preview are still the old ones since absolute links are used instead of relative ones. May need to update this as well

hi @yucheng11122017, what about the `usecase.puml` diagram? It is broken after the dependency upgrade as seen [here](https://github.com/MarkBind/markbind/pull/2311/files#diff-16814c206c8aff01d4ad6090fb91f89adcd6c2c80a3eb18552a43b3268906a14).

lesterong commented:
> hi @lesterong! thanks for the work! perhaps it would be good to check out how the [cs2103 website](https://github.com/nus-cs2103-AY2223S2/website) would look after the bump as it is one of our most content heavy websites which is heavily used. feel free to ping me if you need any help!

hi @itsyme, great idea! Upon some initial testing, I also found that the new version of PlantUML seems to be having some issues rendering the underlines if they are specified using the Creole syntax (`__text__`) in class diagrams. Using the HTML syntax (`<u>text</u>`) appears to work fine. 
```
@startuml
skinparam classAttributeIconSize 0
skinparam shadowing false
hide circle
hide empty members
class a as "__adam:Student__"
class e as "__eve:Student__"
class h as "__h1:Hall__"
class d1 as "__history:Degree__"
class d2 as "__english:Degree__"
a -- d1
a -right- h
e -left- h
e -- "major" d2
@enduml
```

**Before** 
![image](https://github.com/MarkBind/markbind/assets/84223259/a35f864d-6937-4cc3-b9ad-ba1cbe79aaea)

**After**
![image](https://github.com/MarkBind/markbind/assets/84223259/8ff72497-9975-44a9-8db7-ae19c55aba15)

Using the Creole syntax with object diagrams seems to work fine as well.

tlylt commented:
> > LGTM! Minor point for anyone reviewing: the images in the deploy preview are still the old ones since absolute links are used instead of relative ones. May need to update this as well
> 
> hi @yucheng11122017, what about the `usecase.puml` diagram? It is broken after the dependency upgrade as seen [here](https://github.com/MarkBind/markbind/pull/2311/files#diff-16814c206c8aff01d4ad6090fb91f89adcd6c2c80a3eb18552a43b3268906a14).

> though this can be fixed by providing an alias.

Does it work if you apply the fix that you mentioned?

lesterong commented:
> Does it work if you apply the fix that you mentioned?

Hi @tlylt, yes it does. For example, we can provide an alias `c` to `checkout`.

```puml
@startuml
left to right direction
skinparam packageStyle rectangle
actor customer
actor clerk
'alias provided for checkout
rectangle checkout as c {
  customer -- (checkout)
  (checkout) .> (payment) : include
  (help) .> (checkout) : extends
  (checkout) -- clerk
}
@enduml
```

The diagram generated will be as follows:
<img width="343" alt="image" src="https://github.com/MarkBind/markbind/assets/84223259/5c1e65c9-f388-4ac4-bb96-401aa1f50f44">

Let me know if this is the desired workaround.

tlylt commented:
> 



> > Does it work if you apply the fix that you mentioned?
> 
> Hi @tlylt, yes it does. For example, we can provide an alias `c` to `checkout`.
> 
> ```wsd
> @startuml
> left to right direction
> skinparam packageStyle rectangle
> actor customer
> actor clerk
> 'alias provided for checkout
> rectangle checkout as c {
>   customer -- (checkout)
>   (checkout) .> (payment) : include
>   (help) .> (checkout) : extends
>   (checkout) -- clerk
> }
> @enduml
> ```
> 
> The diagram generated will be as follows: <img alt="image" width="343" src="https://user-images.githubusercontent.com/84223259/250297461-5c1e65c9-f388-4ac4-bb96-401aa1f50f44.png">
> 
> Let me know if this is the desired workaround.

Yes, that would suffice. Could you draft up a migration guide for this PlantUML update in the PR description, highlighting the noticeable changes and the required settings/changes needed to handle cases such as the above? This would greatly help current users and will be included in the release note.

lesterong commented:
> Yes, that would suffice. Could you draft up a migration guide for this PlantUML update in the PR description, highlighting the noticeable changes and the required settings/changes needed to handle cases such as the above? This would greatly help current users and will be included in the release note.

Hi @tlylt, I have drafted up a migration guide in the PR description, do take a look and let me know if there are any changes to be made! 🙂

tlylt commented:
> Hi @tlylt, I have drafted up a migration guide in the PR description, do take a look and let me know if there are any changes to be made! 🙂

Thanks @lesterong, looks good to me. Could you help with:
- [ ] add in relevant screenshots into the migration guide so that it is clearer what important visual changes are there
- [ ] I saw that in the latest plantuml version, there's a possibility to download a version without GraphViz. Could you investigate this further and double check on https://plantuml.com/graphviz-dot to find out if this GraphViz dependency should still be mentioned and any new updated information should be included in our documentation? I briefly looked through and it seems like there are now 1. an alternative and 2. doesn't seem to need it on Windows anymore

lesterong commented:
> Thanks @lesterong, looks good to me. Could you help with:
> 
> * [x]  add in relevant screenshots into the migration guide so that it is clearer what important visual changes are there
> * [ ]  I saw that in the latest plantuml version, there's a possibility to download a version without GraphViz. Could you investigate this further and double check on https://plantuml.com/graphviz-dot to find out if this GraphViz dependency should still be mentioned and any new updated information should be included in our documentation? I briefly looked through and it seems like there are now 1. an alternative and 2. doesn't seem to need it on Windows anymore

Hi @tlylt, I have added in some screenshots for the migration guide, do take a look and let me know if there are any changes to be made! 

As for the GraphViz dependency, I'm still playing around and testing with the version that uses Smetana (a Java port of Dot). [Here's the link](https://plantuml.com/smetana02) to PlantUML's Smetana page for reference.

tlylt commented:
> As for the GraphViz dependency, I'm still playing around and testing with the version that uses Smetana (a Java port of Dot). [Here's the link](https://plantuml.com/smetana02) to PlantUML's Smetana page for reference.

Do help to check if the use of Smetana has any limitations or undesirable changes. If it's all good, we can consider passing in that directive when we invoke Plantuml to have the layout engine use Smetana by default

lesterong commented:
> > As for the GraphViz dependency, I'm still playing around and testing with the version that uses Smetana (a Java port of Dot). [Here's the link](https://plantuml.com/smetana02) to PlantUML's Smetana page for reference.
> 
> Do help to check if the use of Smetana has any limitations or undesirable changes. If it's all good, we can consider passing in that directive when we invoke Plantuml to have the layout engine use Smetana by default

Hello @tlylt, I did some more testing with Smetana and it seems like there are still some issues with Smetana, particularly with arrows and nested layouts.

For example, the `state.puml` does not render correctly using Smetana.

**With GraphViz Dot**

<img width="400" src="https://github.com/MarkBind/markbind/assets/84223259/58c71d92-788f-4c73-ac0d-7804b0beda34">


**With Smetana**

<img width="400" src="https://github.com/MarkBind/markbind/assets/84223259/2015a373-88ec-468a-badd-e228964dc2ee">


Another example would be the `component.puml` diagram.

**With GraphViz Dot**

<img width="400" src="https://github.com/MarkBind/markbind/assets/84223259/30858dea-fdbe-49b7-ad08-1630c3d21c7f">


**With Smetana**

<img width="400" src="https://github.com/MarkBind/markbind/assets/84223259/956139c8-5ade-47bb-b50d-6f035137b07b">


Similar issues are also reported in PlantUML's repo:

- plantuml/plantuml#1068
- plantuml/plantuml#1236
- plantuml/plantuml#1408
- plantuml/plantuml#1441
- plantuml/plantuml#1443

tlylt commented:
> Hello @tlylt, I did some more testing with Smetana and it seems like there are still some issues with Smetana, particularly with arrows and nested layouts.

Thanks for checking, @lesterong! In that case, I see no reason to update our setup to use Smetana by default. Could you help me with adding a side note on that alternative (and the issues around it - just the high level description in a few sentences) in https://markbind-master.netlify.app/userguide/components/imagesanddiagrams#diagrams ?

Other than that will get this PR merged soon 👍

lesterong commented:
> > Hello @tlylt, I did some more testing with Smetana and it seems like there are still some issues with Smetana, particularly with arrows and nested layouts.
> 
> Thanks for checking, @lesterong! In that case, I see no reason to update our setup to use Smetana by default. Could you help me with adding a side note on that alternative (and the issues around it - just the high level description in a few sentences) in https://markbind-master.netlify.app/userguide/components/imagesanddiagrams#diagrams ?
> 
> Other than that will get this PR merged soon 👍

Great @tlylt! I have added a bullet point on Smetana [here](https://github.com/MarkBind/markbind/pull/2311/files#diff-3e5850f57a68a8874121c5753282b8bb63705c6abd271c3f19e661b00d97b2ec). Do take a look and let me know if there are any changes required!

tlylt reviewed (approved):
LGTM. @damithc do you have any comments/concerns with the diagrams generated by the new version? If not will be merging this PR in.

damithc commented on docs/userGuide/diagrams/usecase.puml:
```diff
@@ -3,7 +3,7 @@ left to right direction
 actor customer
 actor clerk
-rectangle checkout {
+rectangle checkout as c {
```
```suggestion
rectangle shop {
```

Perhaps it would be less confusing if we give the system a different name?

lesterong replied:
sounds good! I have fixed it up 👍🏼

damithc reviewed (approved):
LGTM. Thanks for the good work @lesterong and for the reviews (@tlylt @yucheng11122017). Just a minor comment only.

tlylt commented:
> * [ ]  I saw that in the latest plantuml version, there's a possibility to download a version without GraphViz. Could you investigate this further and double check on [plantuml.com/graphviz-dot](https://plantuml.com/graphviz-dot) to find out if this GraphViz dependency should still be mentioned and any new updated information should be included in our documentation? I briefly looked through and it seems like there are now 1. an alternative and 2. doesn't seem to need it on Windows anymore

@lesterong sorry I think this item is not completely done: 

> Installation under Windows
Starting from 1.2020.21
If you use a recent version (that is at least version 1.2020.21), you don't need to manually install GraphViz anymore !
A [minimalistic graphviz dot.exe](https://github.com/plantuml/graphviz-distributions) is packed into PlantUML and will be automagically unzipped in some temporary folder if needed (that is, if no installed GraphViz is available).
This is really the prefered option under Windows.
Caveat: Before 1.2020.25, there was an error message during graph generation, so please use 1.2020.25 or more recent.

- from https://plantuml.com/graphviz-dot#:~:text=Installation%20under%20Windows

On top of the alternative layout engine, the new version also brings the possibility of omitting the Graphviz installation on Windows machines. Could you verify?

Also, curious how did you generate and update the images in UG? I think this process is not captured in our DG.

For the above, we may need to update the docs if necessary (hopefully this is the last thing to do in this PR :)

lesterong commented:
> On top of the alternative layout engine, the new version also brings the possibility of omitting the Graphviz installation on Windows machines. Could you verify?

My apologies, I will check on it by this weekend with a Windows PC, and update on it.

> Also, curious how did you generate and update the images in UG? I think this process is not captured in our DG.
> 
> For the above, we may need to update the docs if necessary (hopefully this is the last thing to do in this PR :)

I generated the diagrams using the PlantUML plugin in IntelliJ.

tlylt commented:
> I generated the diagrams using the PlantUML plugin in IntelliJ.

Hmm...best to generate them via MarkBind as well, to verify that it's actually working.

lesterong commented:
> > I generated the diagrams using the PlantUML plugin in IntelliJ.
> 
> Hmm...best to generate them via MarkBind as well, to verify that it's actually working.

I did serve the site locally to test as well, I used the plugin mainly for convenience. But I will double check on this to be sure! 👍🏼

lesterong commented:
As of 12 July 2023, it seems like PlantUML has just released V1.2023.10. Do I upgrade to that version, or leave it at V1.2023.9? The change log from PlantUML is linked [here](https://plantuml.com/changes).

<img width="724" alt="image" src="https://github.com/MarkBind/markbind/assets/84223259/17bac9ce-e39b-45d3-99eb-b1cc72b6ce96">

tlylt commented:
> As of 12 July 2023, it seems like PlantUML has just released V1.2023.10. Do I upgrade to that version, or leave it at V1.2023.9? The change log from PlantUML is linked [here](https://plantuml.com/changes).
> 
> <img alt="image" width="724" src="https://user-images.githubusercontent.com/84223259/253741351-17bac9ce-e39b-45d3-99eb-b1cc72b6ce96.png">

Sure @lesterong if it doesn't take too much time (need to verify that things still work), ideally I would like the PR to be ready soon or by tmr, to make the cut for v5 release,

lesterong commented:
> > As of 12 July 2023, it seems like PlantUML has just released V1.2023.10. Do I upgrade to that version, or leave it at V1.2023.9? The change log from PlantUML is linked [here](https://plantuml.com/changes).
> > <img alt="image" width="724" src="https://user-images.githubusercontent.com/84223259/253741351-17bac9ce-e39b-45d3-99eb-b1cc72b6ce96.png">
> 
> Sure @lesterong if it doesn't take too much time (need to verify that things still work), ideally I would like the PR to be ready soon or by tmr, to make the cut for v5 release.
> 
> Meaning if it is going to take longer than that, would be great if you could wrap up this PR first and keep that as a follow-up item.

No worries, I'll get it up by tomorrow afternoon 🙂.  I'm currently testing it on Windows.

lesterong commented:
> On top of the alternative layout engine, the new version also brings the possibility of omitting the Graphviz installation on Windows machines. Could you verify?

Hi @tlylt, I have verified that on Windows, PlantUML works without a Graphviz installation. I used a Windows 11 machine without a Graphviz Dot installation, and tested it using `markbind serve -d` on the docs. With 2023.1.10, it now generates a the correct image instead of the error image.

<img width="272" alt="image" src="https://github.com/MarkBind/markbind/assets/84223259/6b73aebf-f4d4-497e-9a57-47c72df2716c">

I have also re-tested generating all the `.puml` files with version 2023.1.10 instead of 2023.1.9 using `markbind serve -d`, and I do not see any other issues.

> For the above, we may need to update the docs if necessary (hopefully this is the last thing to do in this PR :)

With that said, do I go ahead and update the user and developer guides to specify that a Graphviz installation is not required for Windows?

tlylt commented:
> I have also re-tested generating all the `.puml` files (both within markbind docs and the se-edu book) with version 2023.1.10 instead of 2023.1.9 using `markbind serve -d`, and I do not see any other issues.
> 
> > For the above, we may need to update the docs if necessary (hopefully this is the last thing to do in this PR :)
> 
> With that said, do I go ahead and update the user and developer guides to specify that a Graphviz installation is not required for Windows?

Thank you. Could you update the docs to mention that & in dev docs add in the steps you took to generate and update the puml images properly? (Under Development -> Workflow -> Dependency management, add at the end of that)

lesterong commented:
> > I have also re-tested generating all the `.puml` files (both within markbind docs and the se-edu book) with version 2023.1.10 instead of 2023.1.9 using `markbind serve -d`, and I do not see any other issues.
> > > For the above, we may need to update the docs if necessary (hopefully this is the last thing to do in this PR :)
> > 
> > 
> > With that said, do I go ahead and update the user and developer guides to specify that a Graphviz installation is not required for Windows?
> 
> Thank you. Could you update the docs to mention that & in dev docs add in the steps you took to generate and update the puml images properly (basically how-to guide for upgrading puml) ? (Under Development -> Workflow -> Dependency management, add at the end of that)

Sure. I will do up a draft by tonight, so we can get it sorted out by tomorrow. 👍🏼

damithc commented:
> Thank you. Could you update the docs to mention that & in dev docs add in the steps you took to generate and update the puml images properly (basically how-to guide for upgrading puml) ? (Under Development -> Workflow -> Dependency management, add at the end of that)

Guys, is this regarding the UML diagrams used to illustrate the image annotation feature? In that case, we can just keep using the same images as before (i.e., they don't need to be updated every time PlantUML is updated) because that feature has nothing to do with PlantUML, right?

lesterong commented:
> > Thank you. Could you update the docs to mention that & in dev docs add in the steps you took to generate and update the puml images properly (basically how-to guide for upgrading puml) ? (Under Development -> Workflow -> Dependency management, add at the end of that)
> 
> Guys, is this regarding the UML diagrams used to illustrate the image annotation feature? In that case, we can just keep using the same images as before (i.e., they don't need to be updated every time PlantUML is updated) because that feature has nothing to do with PlantUML, right?

Correct me if I'm wrong, but I believe @tlylt is referring to all the diagrams in the user guide, not just the annotation diagrams, i.e. all the png images that has been updated in this PR.

tlylt commented:
> > Thank you. Could you update the docs to mention that & in dev docs add in the steps you took to generate and update the puml images properly (basically how-to guide for upgrading puml) ? (Under Development -> Workflow -> Dependency management, add at the end of that)
> 
> Guys, is this regarding the UML diagrams used to illustrate the image annotation feature? In that case, we can just keep using the same images as before (i.e., they don't need to be updated every time PlantUML is updated) because that feature has nothing to do with PlantUML, right?

@damithc we reused two UML diagrams for image annotation, which result in this coupling. I think would be good to decouple it for the reason you mentioned.

@lesterong could you undo the annotation.md changes and copy the two UML used (and rename them) and put them in docs/images (so that they are tgt with annotateSampleImage.png). Update the link in annotation page accordingly.

lesterong commented:
> @lesterong could you undo the annotation.md changes and copy the two original UML images used (and rename them) and put them in docs/images (so that they are tgt with annotateSampleImage.png). Update the link in annotation page accordingly.

Hi @tlylt, I have reverted the changes in `annotations.md`, added two images `annotateSampleObject.png` and `annotateSampleSequence.png`, and updated `annotations.md` to point to the two images. 

> Thank you. Could you update the docs to mention that & in dev docs add in the steps you took to generate and update the puml images properly (basically how-to guide for upgrading puml) ? (Under Development -> Workflow -> Dependency management, add at the end of that)

I have also updated the user guide and developer guides to point out that Graphviz is optional in Windows:

**Updated user guide `imagesAndDiagrams.md`**
<img width="811" alt="image" src="https://github.com/MarkBind/markbind/assets/84223259/034c8220-8a14-4081-a27c-7ba0260dacaf">

**Updated developer guide `settingUp.md`**
<img width="818" alt="image" src="https://github.com/MarkBind/markbind/assets/84223259/bacab49b-30a6-490e-ae06-81c86629f4a1">

- Not sure if it's better to split up the Java and Graphviz into two separate bullet points, or is this sufficient?


And also added a section on updating PlantUML in the workflows page

<img width="822" alt="image" src="https://github.com/MarkBind/markbind/assets/84223259/4f97ef5a-74e7-4396-909a-00b31c53ebe1">

Do take a look and let me know if there are any changes to be made. Thanks!

tlylt commented:
> And also added a section on updating PlantUML in the workflows page

@lesterong this change is not pushed? I don't see it anywhere.

lesterong commented:
> 

My apologies, will push it now

tlylt reviewed (approved):
LGTM, thank you @lesterong 🚀
